tests/interpolation_test.py

```diff
@@ -0,0 +1,83 @@
+# coding=UTF-8
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (print_function, division, absolute_import,
+                        unicode_literals)
+import codecs
+import difflib
+import os.path
+import shutil
+import sys
+import tempfile
+import unittest
+
+import defcon
+from glyphsLib.builder import GLYPHS_PREFIX
+from glyphsLib.interpolation import build_designspace
+
+
+def makeFamily(familyName):
+    m1, m2 = defcon.Font(), defcon.Font()
+    m1.info.familyName, m1.info.styleName = familyName, "Regular"
+    m1.lib[GLYPHS_PREFIX + "weightValue"] = 400.0
+    m2.info.familyName, m2.info.styleName = familyName, "Black"
+    m2.lib[GLYPHS_PREFIX + "weightValue"] = 900.0
+    instances = {
+        "defaultFamilyName": familyName,
+        "data": [
+            {"name": "Regular", "interpolationWeight": 400.0},
+            {"name": "Semibold", "interpolationWeight": 600.0},
+            {"name": "Bold", "interpolationWeight": 700.0},
+            {"name": "Black", "interpolationWeight": 900.0},
+        ],
+    }
+    return [m1, m2], instances
+
+
+class DesignspaceTest(unittest.TestCase):
+    def expect_designspace(self, masters, instances, expectedFile):
+        master_dir = tempfile.mkdtemp()
+        designspace, _ = build_designspace(
+            masters, master_dir, os.path.join(master_dir, "out"), instances)
+        with codecs.open(designspace, "r", "utf-8") as f:
```

Comment:
`io.open` is the recommended way to open files in py2.py3, `codecs.open` is kept in python3 only for backward compatibility.